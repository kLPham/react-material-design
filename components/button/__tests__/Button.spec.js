import React from 'react';
import { shallow } from 'enzyme';
import Button from '../index';

describe('<Button />', () => {
    it('should render flat', () => {
        const label = 'Flat button';
        const button = shallow(
            <Button label={label} />,
        );
        expect(button).toMatchSnapshot();
    });
    it('shound render darkTheme', () => {
        const label = 'Flat button';
        const darkTheme = true;
        const button = shallow(
            <Button label={label} darkTheme={darkTheme} />,
        );
        expect(button).toMatchSnapshot();
    });
    it('should render colored', () => {
        const label = 'Colored button';
        const colored = true;
        const button = shallow(
            <Button colored={colored} label={label} />,
        );
        expect(button).toMatchSnapshot();
    });

    it('should render raised', () => {
        const label = 'Raised button';
        const raised = true;
        const button = shallow(
            <Button raised={raised} label={label} />,
        );
        expect(button).toMatchSnapshot();
    });

    it('should render raised disabled', () => {
        const label = 'Raised disabled button';
        const raised = true;
        const disabled = true;
        const button = shallow(
            <Button raised={raised} label={label} disabled={disabled} />,
        );
        expect(button).toMatchSnapshot();
    });

    it('should render dense', () => {
        const label = 'Dense button';
        const dense = true;
        const button = shallow(
            <Button label={label} dense={dense} />,
        );
        expect(button).toMatchSnapshot();
    });

    it('should render compact', () => {
        const label = 'Compact button';
        const compact = true;
        const button = shallow(
            <Button label={label} compact={compact} />,
        );
        expect(button).toMatchSnapshot();
    });

    it('should render primary', () => {
        const label = 'Primary button';
        const primary = true;
        const button = shallow(
            <Button label={label} primary={primary} />,
        );
        expect(button).toMatchSnapshot();
    });
});
