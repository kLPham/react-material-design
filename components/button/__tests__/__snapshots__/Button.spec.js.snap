// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render colored 1`] = `
<button
  className="mdc-button mdc-button--accent"
>
  Colored button
</button>
`;

exports[`<Button /> should render compact 1`] = `
<button
  className="mdc-button mdc-button--compact"
>
  Compact button
</button>
`;

exports[`<Button /> should render dense 1`] = `
<button
  className="mdc-button mdc-button--dense"
>
  Dense button
</button>
`;

exports[`<Button /> should render flat 1`] = `
<button
  className="mdc-button"
>
  Flat button
</button>
`;

exports[`<Button /> should render primary 1`] = `
<button
  className="mdc-button mdc-button--primary"
>
  Primary button
</button>
`;

exports[`<Button /> should render raised 1`] = `
<button
  className="mdc-button mdc-button--raised"
>
  Raised button
</button>
`;

exports[`<Button /> should render raised disabled 1`] = `
<button
  className="mdc-button mdc-button--raised"
  disabled={true}
>
  Raised disabled button
</button>
`;
